### PR TITLE
Use transitions for Dismissable

### PR DESCRIPTION
**Pull request by mpcomplete**
Simplifies the code a bit.

### CODE REVIEW

File: sky/packages/sky/lib/widgets/dismissable.dart [hunk 1]
```diff
@@ -91,38 +83,27 @@ class Dismissable extends AnimatedComponent {
...
 
   void _handlePointerDown(sky.PointerEvent event) {
-    setState(() {
-      _dragUnderway = true;
-      _dragX = 0.0;
-      _fadePerformance.progress = 0.0;
-    });
+    _dragUnderway = true;
+    _dragX = 0.0;
+    _fadePerformance.progress = 0.0;
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/dismissable.dart]: Why no setState here?  Don't we need to schedule this component to be built?

File: sky/packages/sky/lib/widgets/dismissable.dart [hunk 2]
```diff
@@ -166,24 +140,29 @@ class Dismissable extends AnimatedComponent {
...
 
   Widget build() {
     if (_resizePerformance != null) {
-      AnimatedValue<double> height = _resizePerformance.variable;
-      return new Container(height: height.value);
+      AnimatedValue<double> dismissHeight = new AnimatedValue<double>(
+        _size.height,
+        end: 0.0,
+        curve: ease,
+        interval: new Interval(_kCardDismissResizeDelay, 1.0)
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/dismissable.dart]: s/new/const/  ?

File: sky/packages/sky/lib/widgets/transitions.dart [hunk 3]
```diff
@@ -167,3 +169,42 @@ class ColorTransition extends TransitionBase {
...
+  AnimatedValue<double> width;
+  AnimatedValue<double> height;
+
+  void syncFields(SquashTransition updated) {
+    width = updated.width;
+    height = updated.height;
+    super.syncFields(updated);
+  }
+
+  double _maybe(AnimatedValue<double> x) => x != null ? x.value : null;
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/transitions.dart]: This feels out-of-place here...  maybe make this into a free function at the top of this file?

**Comment by abarth** [comment]
LGTM

File: sky/packages/sky/lib/widgets/dismissable.dart (hunk 1 above)
Comment (mpcomplete) [inline sky/packages/sky/lib/widgets/dismissable.dart]: The slide/fade transitions rebuild themselves as the _fadePerformance changes. The rest of the component doesn't need to rebuild.

File: sky/packages/sky/lib/widgets/dismissable.dart (hunk 2 above)
Comment (mpcomplete) [inline sky/packages/sky/lib/widgets/dismissable.dart]: You mean the Interval? Done.

File: sky/packages/sky/lib/widgets/transitions.dart (hunk 3 above)
Comment (mpcomplete) [inline sky/packages/sky/lib/widgets/transitions.dart]: Done.

**Pull request merged**